dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/InteractionPredicate.kt

```diff
@@ -0,0 +1,21 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.tracking
+
+import com.datadog.tools.annotation.NoOpImplementation
+
+/**
+ * Provides custom attributes for the events produced by the user action tracking strategy.
+ */
+@NoOpImplementation
+interface InteractionPredicate {
```

Comment:
normally predicate means it should output `true` or `false` (or `object` or null) to decide if interaction should be used or not, but here we just use this interface to change the property of interaction. do you think it may cause any confusion for the users? may they think that if they return `null`, then interaction is not recorded at all?

Reply:
Yes I do agree but had to follow what we did for `Views` also -> https://github.com/DataDog/dd-sdk-android/blob/master/dd-sdk-android/src/main/kotlin/com/datadog/android/rum/tracking/ComponentPredicate.kt otherwise will loose the consistency on our SDK configuration. Regarding the return value we explain this in the method docs, we did not have a problem with the `ComponentPredicate`, I think we will be safe here also.

Reply:
ok, let's keep it for the consistency then.